### ISSUE TITLE
Remove Global Variables

### ISSUE THREAD

**Issue by jackolney** [jackolney/CascadeDashboard#3]
According to shinyapps.io support, because multiple users (connections) are made to a single worker (R-process) on a particular instance. If global variables are used anywhere inside a shiny app, they can be leaked between users – this was the root cause of the embarrassment in Croatia.

Need to come up with a decent strategy to ensure that only reactive elements are passed around the tool.

```r
# Perhaps a solution would be:

MasterData <- reactiveValues()

MasterData$someData

# And then just replace the someData elements?
# This should keep things reactive and minimise other problems
```

**Comment by jackolney** [comment]
Scoping information here: http://shiny.rstudio.com/articles/scoping.html

FYI, `MasterData` only needs to be accessible to server.R

**Comment by jackolney** [comment]
To print all global variables: `print(as.list(.GlobalEnv))` 

Maybe include this in a diagnostic script and then work to reduce, delete all global variables? 

Also identify how many environments are open / in existence and identify which are bleeding variables

**Comment by jackolney** [comment]
Some global variables are okay to leave in (I believe) – such as `rhandsontable` global FLAGS

**Comment by jackolney** [comment]
36 global variables to organise:
![screen shot 2017-02-06 at 15 50 39](https://cloud.githubusercontent.com/assets/4134882/22654235/0f627682-ec84-11e6-9773-b3f15a30aa10.png)

**Comment by jackolney** [comment]
### Plan

e.g.
- Set `MasterData <- list()` in `server.R`
- Then in functions I can assign/update values with `<<-`
- This variable will then remain as 'per-session' and not become Global.

**Comment by jackolney** [comment]
## Masterdata
- [x] MasterData

## Calibration
- [ ] CalibIncOut
- [ ] CalibInitOut
- [ ] CalibOut
- [ ] CalibParamOut
- [ ] countryReportName
- [ ] countryReportName
- [ ] minError
- [ ] minErrorRun
- [ ] ParamMaxMin
- [ ] runError
- [ ] selectedRuns

## Optimisation
- [ ] baseTenPercentCalibInitial
- [ ] shuffledRuns
- [ ] optResults
- [ ] BaselineAdhr
- [ ] BaselineCost
- [ ] BaselineInit
- [ ] BaselineLink
- [ ] BaselinePreR
- [ ] BaselineRetn
- [ ] BaselineTest

## Report
- [ ] report_909090_adherence
- [ ] report_909090_adherence_BASE
- [ ] report_909090_cost
- [ ] report_909090_cost_BASE
- [ ] report_909090_initiation
- [ ] report_909090_initiation_BASE
- [ ] report_909090_linkage
- [ ] report_909090_linkage_BASE
- [ ] report_909090_preRetention
- [ ] report_909090_preRetention_BASE
- [ ] report_909090_retention
- [ ] report_909090_retention_BASE
- [ ] report_909090_testing
- [ ] report_909090_testing_BASE

**Comment by jackolney** [comment]
Complete – need to merge `global` with `master`

**Comment by jackolney** [comment]
Merged in 53cd7c5a0cccbd82fa62a72eba5a9833993e1ca1
